mapie/multi_label_classification.py

```diff
@@ -0,0 +1,738 @@
+    ) -> None:
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+        self.verbose = verbose
+
+    def _check_parameters(self) -> None:
+        """
+        Perform several checks on input parameters.
+
+        Raises
+        ------
+        ValueError
+            If parameters are not valid.
+        """
+        check_n_jobs(self.n_jobs)
+        check_verbose(self.verbose)
+        check_random_state(self.random_state)
+
+    def _check_method(self) -> None:
+        """Check that the specified method is valid
+
+        Raises
+        ------
+        ValueError
+            Raise error if the name of the method is not
+            in ["crc", "rcps"]
+        """
+        if self.method not in self.valid_methods_:
+            raise ValueError(
+                "Invalid method. "
+                "Allowed values are 'crc' or 'rcps"
+            )
+
+    def _check_all_labelled(self, y: NDArray) -> None:
+        """Check that all observations have at least
+        one label
+
+        Parameters
+        ----------
+        y : NDArray of shape (n_samples, n_labels)
+            Labels of the observations.
+
+        Raises
+        ------
+        ValueError
+            Raise error if at least one observation
+            has no label.
+        """
+        if 0 in np.unique(y.sum(axis=1)):
+            raise ValueError(
+                "Invalid y. "
+                "All observations should contain at "
+                "least one label."
+            )
+
+    def _check_delta(self, delta: Optional[float]):
+        """Check that delta is not ``None`` when the
+        method is RCPS and that it has values between
+        0 and 1.
+
+        Parameters
+        ----------
+        delta : float
+            Probability with wchi we control the risk.
+
+        Raises
+        ------
+        ValueError
+            If delta is ``None`` and method is RCSP
+        ValueError
+            If delta value is not in [0, 1] and method
+            is RCPS.
+        """
+        if (self.method == "rcps") and (delta is None):
+            raise ValueError(
+                "Invalid delta. "
+                "delta can not be None when using "
+                "a RCPS method."
+            )
+        if hasattr(delta, "__len__"):
+            raise ValueError(
+                "Invalid delta. "
+                f"delta must be a float, not a {type(delta)}"
+            )
+        if (delta is not None) and (
+            ((delta <= 0) or (delta >= 1)) and
+            self.method == "rcps"
+        ):
+            raise ValueError(
+                "Invalid delta. "
+                "delta must be in the ]0, 1[ interval"
+            )
+
+    def _check_estimator(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        estimator: Optional[ClassifierMixin] = None,
+    ) -> ClassifierMixin:
+        """
+        Check if estimator is ``None``,
```

Comment:
Check ?

Reply:
what is the problem ?

Reply:
changed to 
Check the estimator value. If it is ``None``,
        it returns a multi-output